proposals/0029-cooperative-vector.md

```diff
@@ -102,383 +113,327 @@ Introduce new HLSL intrinsics to accelarate matrix-vector operations. In this sp
-template<typename DESC, typename InputTy, typename ResultTy, uint InputComponents>
-vector<ResultTy, DESC::M> VectorMatrixMulAdd(vector<InputTy, InputComponents> InputVector,
-                               (RW)ByteAddressBuffer Matrix,
-                               uint MatrixOffset,
-                               uint MatrixStride,
-                               (RW)ByteAddressBuffer BiasVector,
-                               uint BiasOffset);
-
-// Result = Matrix * InputVector
-template<typename DESC, typename InputTy, typename ResultTy, uint InputComponents>
-vector<ResultTy, DESC::M> VectorMatrixMul(vector<InputTy, InputComponents> InputVector,
-                             (RW)ByteAddressBuffer Matrix,
-                             uint MatrixOffset,
-                             uint MatrixStride);
+The `@dx.op.matvecmul` operation multiplies a **MxK** dimension matrix and a
+**K** sized input vector. The matrix is loaded from memory while the vector is
+stored in a variable.
 
-```
+The `@dx.op.matvecmuladd` operation behaves as `@dx.op.matvecmul`, but also adds
+an **M**-sized bias vector (loaded from memory) to the result.
 
-*InputVector* is the vector operand of the matrix-vector mul/mul-add operation. *InputTy* is the physical storage type
- of the elements of the vector, which might vary from the actual type that the elements of the vector are interpreted
- as, *InputInptretation* from *DESC*. *InputComponents* is the number of components in the input vector, which equals
- the matrix dimension *K* for a non-packed type and for a packed type, equals the least number that can hold *K* values
- of the packed type. Where, packed type, refers to types like `SignedInt8x4Packed` where each 32-bit element of the
- vector corresponds to four 8-bit signed integers; Unpacked types are the standard types like float16, uint etc. The
- elements of the *InputVector* are converted to type specified by *DESC: Ii* present in, if it is legal. More details
- in the [Type Interpretations](#type-interpretations) section.
+> Note that the dimensions of the matrix are **M**x**K** versus **M**x**N**
+> usually found in linear algebra textbooks. This is to futureproof for
+> potential matrix-matrix operations in the future where the inputs could be
+> **M**x**K** and **K**x**N** to produce an **M**x**N** result matrix.
 
-*Matrix* is loaded starting from a byte offset *MatrixOffset* from the start of Buffer, and raw data is loaded according
- to the type interpretation parameter *DESC: Mi*. *DESC: MxK* is the dimension of the matrix. No conversion is
- performed. The *MatrixOffset* and the base address of the Matrix buffer must be 64B aligned. The *DESC: Layout* of the
- matrix is one of the enum values *DXILMatrixLayout* listed above.
+#### Arguments
 
-*MatrixStride*, for RowMajor or ColumnMajor layouts, is the number of bytes to go from one row/column to the next. For
- optimal layouts, *MatrixStride* is ignored.
+| Argument | Type | 
```

Comment:
Missed this, what was intended in this table?

Reply:
Oh gosh, thank you!  That was going to be a list of arguments saying which ones were immarg or not, but then I found the immarg notation that the LLVM documentation uses and so I just switched to that - and forgot to remove the table!